File: app/src/main/java/com/dmi/perfectreader/util/android/Units.java
package com.dmi.perfectreader.util.android;

import android.content.Context;
import android.util.DisplayMetrics;

public class Units {
    private static float density;

    public static void init(Context context) {
        DisplayMetrics displayMetrics = context.getApplicationContext().getResources().getDisplayMetrics();
        density = displayMetrics.density;
    }

    public static float dipToPx(float dip) {
        return dip * density;
    }

    public static float pxToDip(float px) {
        return px / density;
    }
}

File: util/util-android/src/main/kotlin/com/dmi/util/Units.kt
package com.dmi.util

import android.content.Context
import android.util.DisplayMetrics

object Units {
    private var density = 0F

    fun init(context: Context) {
        val displayMetrics = context.applicationContext.resources.displayMetrics
        density = displayMetrics.density
    }

    fun dipToPx(dip: Float): Float {
        return dip * density
    }

    fun pxToDip(px: Float): Float {
        return px / density
    }
}